### PR TITLE
Add gibon w/ npm auto-update via single package.json

### PR DESCRIPTION
**Pull request by dakshshah96**
Pull request for issue: #10651 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/tunnckoCore/gibon
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/gibon
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 15
   - Count of stars: 392
   - Count of forks: 12
 * NPM download stats (optional):
   - Downloads in the last day: 1
   - Downloads in the last week: 17
   - Downloads in the last month: 253

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [ ] Auto-update setup
 * [ ] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by PeterDaveHello** [comment]
@dakshshah96 please take a look at https://ci.cdnjs.com/cdnjs/cdnjs/4758

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by dakshshah96** [comment]
What correction is needed to the PR?

**Comment by dakshshah96** [comment]
@pvnr0082t I've updated by PR 👍

File: ajax/libs/gibon/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "vanilla",
+    "virtual-dom",
+    "virtual"
+  ],
+  "npmName": "gibon",
+  "npmFileMap": [
+    {
+      "basePath": "dist",
+      "files": [
+        "**/*"
```
Comment (pvnr0082t) [inline ajax/libs/gibon/package.json]: We should exclude `*.gz` files. Let's modify it to `"**/*.!(*gz)"`. Thanks.

File: ajax/libs/gibon/package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "gibon",
+  "version": "0.3.2",
+  "description": "Functional client-side router in ~570 bytes, built on HTML5 History API",
+  "repository": "tunnckoCore/gibon",
```
Comment (pvnr0082t) [inline ajax/libs/gibon/package.json]: Please modify it to
```
"repository": {
  "type": "git",
  "url": "https://github.com/tunnckoCore/gibon"
},
```

File: ajax/libs/gibon/package.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+  "name": "gibon",
+  "version": "0.3.2",
```
Comment (pvnr0082t) [inline ajax/libs/gibon/package.json]: If we add library via single package.json, we should remove version field. Thanks.

**Comment by dakshshah96** [comment]
Thanks for the help @pvnr0082t! I've implemented your suggestions 👍

File: ajax/libs/gibon/package.json [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{
+  "name": "gibon",
+  "description": "Functional client-side router in ~570 bytes, built on HTML5 History API",
+  "repository": {
+  "type": "git",
+  "url": "https://github.com/tunnckoCore/gibon"
```
Comment (pvnr0082t) [inline ajax/libs/gibon/package.json]: wrong indent.

**Comment by dakshshah96** [comment]
@pvnr0082t Done 👍

**Comment by dakshshah96** [comment]
@pvnr0082t, @maruilian11, @PeterDaveHello Any changes needed?

**Pull request merged**